Check that db is selected before loading babelfishpg_tsql

staticlibs opened the pull request:
### Description

When `babelfishpg_tsql` [is being loaded on backend init](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/dce7212f5eff7d149620bf00225b6fba496b886d/src/backend/tcop/postgres.c#L4259) there is a check for extension OID. It appeared that `get_extension_oid` that performs the check can [fail hard](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/dce7212f5eff7d149620bf00225b6fba496b886d/src/backend/utils/cache/relcache.c#L352) with:

```
cannot read pg_class without having selected a database
```

This patch adds a check that DB is selected before calling to `get_extension_oid`.

References:
 - [upstream discussion of a similar problem](https://www.postgresql.org/message-id/4EDAA3B0.1010707%40fuzzy.cz)
 - [similar problem in TimescaleDB extension](https://github.com/timescale/timescaledb/issues/1847)

### Issues Resolved

[#1665](https://github.com/babelfish-for-postgresql/babelfish_extensions/issues/1665)
 
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

staticlibs committed:
Check that db is selected before loading babelfishpg_tsql

Signed-off-by: Alex Kasko <alex@staticlibs.net>

rishabhtanwar29 reviewed (approved):
Changes look good to me. Should we update existing [github action](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/BABEL_3_X_DEV__PG_15_X/.github/workflows/main.yml) with `--enable-tap-tests` option and run `make check world` just to be make sure we don't break anything in future as well?

staticlibs commented:
Please note, that `make check-world` is still failing (on `pg_dump` tests) after this fix applied. `pg_dump` currently does not work with non-Babelfish databases, unlike `pg_dumpall` ([link](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/f492997643b8584bf7e5cef670fb41727199c600/src/bin/pg_dump/pg_dumpall.c#L747)) calls to Babelfish catalogs logic in `pg_dump` are not completely guarded by some kind of `is_bbf_db` flag.
Edit: just noticed that `pg_dump` is being fixed in #237 .

rishabhtanwar29 commented:
Yep, I am working on the fix https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/237.

rishabhtanwar29 commented:
PR https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/237 has been merged now, I think we can proceed.

staticlibs committed:
Enable check-world on CI

Signed-off-by: Alex Kasko <alex@staticlibs.net>

staticlibs commented:
Rebased the branch to latest and added `check-world` run on CI:

1. `check-world` is [passing on the PR repo](https://github.com/wiltondb/postgresql_modified_for_babelfish/actions/runs/6613111324/job/17960305431)
2. added `--with-icu` and `--enable-cassert` to configure just in case, not sure if desired
3. disabled parallel test run, its output appeared to be hard to discern, the run takes about 12 minutes now
4. left `upload-test-summary` step as is, not sure if it is possible easily upload all the details from `check-world`

rishabhtanwar29 commented:
Looks good to me.
